test/parallel_api/ranges/std_ranges_zip_view_algo.pass.cpp

```diff
@@ -0,0 +1,42 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "std_ranges_test.h"
+
+std::int32_t
+main()
+{
+#if _ENABLE_STD_RANGES_TESTING
+    using namespace test_std_ranges;
+    namespace dpl_ranges = oneapi::dpl::ranges;
+
+    const int n = medium_size;
+
+    auto zip_proj = [](auto&& val) ->decltype(auto) { return std::get<0>(val); };
+
+    //find_if with zip_view
+    test_range_algo<0>{n}.test_view(dpl_ranges::views::zip, dpl_ranges::find_if, std::ranges::find_if, pred, zip_proj);
+
+    //sort with zip_view
+    test_range_algo<1>{n}.test_view(dpl_ranges::views::zip, dpl_ranges::sort, std::ranges::sort, std::less{}, zip_proj);
```

Comment:
As far as I understand, there is only one underlying sequence.

Could you add a test case, which tests multiple sequences on the device, e.g. imitating sort_by_key, and then checking both keys and values? I believe that the tests are incomplete without it.

Reply:
It is impossible with `test_range_algo`... It needs to another "tester" or "full" test code. It needs to think about.

Reply:
I guess it this case is specific to `zip_view` and there is no need to extend `test_range_algo`.

We can embed that case right here, similarly to what you have already done in `test/parallel_api/ranges/std_ranges_zip_view.pass.cpp` for host policies.

Reply:
I think such case can be added as a new case into `sort_by_key.pass` test.

Reply:
I see that now there is such a check in [std_ranges_zip_view.pass.cpp](https://github.com/uxlfoundation/oneDPL/pull/1877/files#diff-3916d10d53ec41fc870585686848f6de0d2ab116cf8ddad0ca3abebf3919d436).